lib/fs/lstat_windows.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2015 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// +build windows
+
+package fs
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func fixLongPath(path string) string {
+	//TODO: extract the implementation from https://golang.org/src/os/path_windows.go
+	//Probably not necessary, as all walked paths are Canonicalize()-d.
```

Comment:
Yeah, since we do the `\\?\` dance already on Windows this should be unnecessary. I was thinking of removing that since Go does the same internally, but since we're now relying on that to pass things to syscalls directly I guess it needs to stay indefinitely.